src/log.h

```diff
@@ -0,0 +1,25 @@
+/* copyright 2013 Sascha Kruse and contributors (see LICENSE for licensing information) */
+
+#include <glib.h>
+#include <stdbool.h>
+#include <stdlib.h>
+
+#ifndef DUNST_LOG_H
+#define DUNST_LOG_H
+
+#define LOG_E g_error
+#define LOG_C g_critical
+#define LOG_W g_warning
+#define LOG_M g_message
+#define LOG_I g_info
+#define LOG_D g_debug
+
+#define die(...) {LOG_C(__VA_ARGS__); exit(EXIT_FAILURE);}
```

Comment:
It's usual to use a `do { } while(0)` clause in macros instead of simple brackets. See [here](https://stackoverflow.com/questions/154136/why-use-apparently-meaningless-do-while-and-if-else-statements-in-macros) for an explanation.

Since this is a macro that can be used pretty much anywhere in the code I think it's worth making sure it's working as expected.

---

Also, since we're converting `die` to a macro shouldn't it be capitalized?

Reply:
@tsipinakis I've seen this quite often. But I don't get what the actual advantage over a simple scope is. I know, you have to scope macros, that they won't change the programming context, but I'm missing the point, why this has to be done with `do .. while`.

Could you please explain this to me? The answers in the stackoverflow don't really enlighten me.

Reply:
I've created a program and tested the actual difference. Is it really the case, that you want to have `do ... while` just to assert that the macro will have a trailing semicolon?

Reply:
Consider this code
```c
#define foo(x) {a(x); b();}

...

if (some_bool)
       foo(some_bool);
else
...
```
It looks like it works and it's similar to our `die()` macro which is pretty likely to be used under an if statement.

But it doesn't, after expansion we get this:

```c
if (some_bool)
    {a(some_bool); b();}; <---- SYNTAX ERROR, trailing semicolon
else 
   ...
```

The correct form would've been to not use a semicolon since it's a macro but IMO it's better to make it look like actual code and not have corner cases like that.

```c
if (some_bool)
        do { a(some_bool); b();} while(0); <-- Semicolon is supposed to be there
else
...
```